nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/service/nifi-common.service.ts

```diff
@@ -171,4 +187,230 @@ export class NiFiCommon {
+        let days: number = duration / NiFiCommon.MILLIS_PER_DAY;
+        days = days >= 1 ? parseInt(`${days}`, 10) : 0;
+        duration %= NiFiCommon.MILLIS_PER_DAY;
+
+        // remaining duration should be less than 1 day, get number of hours
+        let hours = duration / NiFiCommon.MILLIS_PER_HOUR;
+        hours = hours >= 1 ? parseInt(`${hours}`, 10) : 0;
+        duration %= NiFiCommon.MILLIS_PER_HOUR;
+
+        // remaining duration should be less than 1 hour, get number of minutes
+        let minutes = duration / NiFiCommon.MILLIS_PER_MINUTE;
+        minutes = minutes >= 1 ? parseInt(`${minutes}`, 10) : 0;
+        duration %= NiFiCommon.MILLIS_PER_MINUTE;
+
+        // remaining duration should be less than 1 minute, get number of seconds
+        let seconds = duration / NiFiCommon.MILLIS_PER_SECOND;
+        seconds = seconds >= 1 ? parseInt(`${seconds}`, 10) : 0;
+
+        // remaining duration is the number millis (don't support sub millisecond resolution)
+        duration = Math.floor(duration % NiFiCommon.MILLIS_PER_SECOND);
+
+        // format the time
+        const time =
+            this.pad(hours, 2, '0') +
+            ':' +
+            this.pad(minutes, 2, '0') +
+            ':' +
+            this.pad(seconds, 2, '0') +
+            '.' +
+            this.pad(duration, 3, '0');
+
+        // only include days if appropriate
+        if (days > 0) {
+            return days + ' days and ' + time;
+        } else {
+            return time;
+        }
+    }
+
+    /**
+     * Formats the specified number of bytes into a human readable string.
+     *
+     * @param {number} dataSize
+     * @returns {string}
+     */
+    public formatDataSize(dataSize: number): string {
+        let dataSizeToFormat: number = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_TERABYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' TB';
+        }
+
+        // check gigabytes
+        dataSizeToFormat = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_GIGABYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' GB';
+        }
+
+        // check megabytes
+        dataSizeToFormat = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_MEGABYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' MB';
+        }
+
+        // check kilobytes
+        dataSizeToFormat = parseFloat(`${dataSize / NiFiCommon.BYTES_IN_KILOBYTE}`);
+        if (dataSizeToFormat > 1) {
+            return dataSizeToFormat.toFixed(2) + ' KB';
+        }
+
+        // default to bytes
+        return parseFloat(`${dataSize}`).toFixed(2) + ' bytes';
+    }
+
+    /**
+     * Formats the specified integer as a string (adding commas). At this
+     * point this does not take into account any locales.
+     *
+     * @param {integer} integer
+     */
+    public formatInteger(integer: number): string {
+        let string: string = integer + '';
```

Comment:
I realize this was from `nf-common`, but I would probably suggest using `String(integer)` over type coercion since it's more clear.